test/simulator/timestamp/call_log_manager/call_log_manager.cpp

```diff
@@ -167,24 +167,42 @@ call_log_manager::call_log_query_timestamp(const json &call_log_entry)
     /* Convert the config char * to a string object. */
     std::string config = call_log_entry["input"]["config"].get<std::string>();
 
-    /*
-     * A generated call log without a configuration string in the set timestamp entry will have the
-     * string "(null)", default to all_durable.
-     */
-    if (config == "(null)")
-        config = "get=all_durable";
-
+    std::string class_name = call_log_entry["class_name"].get<std::string>();
     std::string hex_ts;
     bool ts_supported;
+    int ret;
+    /* Check whether we're querying a global connection timestamp or a session transaction
+     * timestamp. */
+    if (class_name == "connection") {
+        /*
+         * A generated call log without a configuration string in the set timestamp entry will have
+         * the string "(null)", default to all_durable.
+         */
+        if (config == "(null)")
+            config = "get=all_durable";
+
+        ret = _conn->query_timestamp(config, hex_ts, ts_supported);
+    } else if (class_name == "session") {
+        /*
+         * A generated call log without a configuration string in the set timestamp entry will have
+         * the string "(null)", default to read timestamp.
+         */
+        if (config == "(null)")
```

Comment:
Could you double check this part, whether we consider "read" if the config is null?

Reply:
See the above comment.